### PR TITLE
support react select

### PR DESCRIPTION
**Pull request by LawyZheng**
| :rocket: | This description was created by [Ellipsis](https://www.ellipsis.dev) for commit 2b1bce6ffc9462ae3212896e4aa298abdb88636c  | 
|--------|--------|

### Summary:
Added support for handling React Select dropdowns, including new exception classes, utility functions, and updates to existing handlers and DOM utilities.

**Key points**:
- **New Exception**: Added `ElementIsNotReactSelectDropdown` in `skyvern/exceptions.py`.
- **Handler Update**: Modified `handle_select_option_action` in `skyvern/webeye/actions/handler.py` to support React Select dropdowns.
- **DOM Utilities**: Added `isReactSelectDropdown`, `getReactSelectOptionElements`, and `getReactSelectOptions` in `skyvern/webeye/scraper/domUtils.js`.
- **DOM Interaction**: Updated `SkyvernElement` class in `skyvern/webeye/utils/dom.py` to include methods for React Select dropdowns.
- **Frame Interaction**: Added `get_react_select_options` method in `SkyvernFrame` class in `skyvern/webeye/utils/page.py`.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

File: skyvern/webeye/utils/dom.py [hunk 1]
```diff
@@ -411,6 +430,71 @@ async def select_by_index(
         await options.nth(index).click(timeout=timeout)
 
 
+class ReactSelectDropdown(AbstractSelectDropdown):
+    def __init__(self, skyvern_frame: SkyvernFrame, skyvern_element: SkyvernElement) -> None:
```
Comment (wintonzheng) [inline skyvern/webeye/utils/dom.py:434]: this can be moved to the base class

File: skyvern/webeye/utils/dom.py [hunk 2]
```diff
@@ -107,6 +108,16 @@ async def is_select2_dropdown(self) -> bool:
             or (tag_name == "input" and "select2-input" in element_class)
         )
 
+    async def is_react_select_dropdown(self) -> bool:
+        tag_name = self.get_tag_name()
+        element_class = await self.get_attr("class")
+        if element_class is None:
+            return False
+
+        return (tag_name == InteractiveElement.INPUT and "select__input" in element_class) or (
```
Comment (wintonzheng) [inline skyvern/webeye/utils/dom.py:117]: is this also gonna work for the react js in https://bighatbiosciences.pinpointhq.com/en/postings/535636c2-97ac-4efc-ae41-93103507ed64/applications/new?

File: skyvern/webeye/scraper/domUtils.js [hunk 3]
```diff
@@ -687,6 +695,78 @@ async function getSelect2Options(element) {
...
+  while (true) {
+    // sometimes need more time to load the options
+    console.log("wait 5s to load all options");
+    await sleep(5000); // wait 5s
+    optionList = dropdownDiv.querySelectorAll("div[role='option']");
+    if (optionList.length === 0) {
+      break;
+    }
+
+    if (
```
Comment (wintonzheng) [inline skyvern/webeye/scraper/domUtils.js:722]: nice if check!

File: skyvern/webeye/utils/dom.py (hunk 1 above)
Comment (LawyZheng) [reply skyvern/webeye/utils/dom.py:434]: Maybe we can do it if we failed to find a general solution for the select pattern.

**Pull request merged**